src/world/zone_connection.rs

```diff
@@ -1062,18 +1061,28 @@ impl ZoneConnection {
                 Task::SetInnWakeup { watched } => {
                     self.player_data.saw_inn_wakeup = *watched;
                 }
-                Task::UnlockMount { id } => {
+                Task::ToggleMount { id } => {
                     let order;
                     {
                         let mut game_data = self.gamedata.lock().unwrap();
                         order = game_data.find_mount_order(*id).unwrap_or(0);
                     }
 
+                    let (value, index) = value_to_flag_byte_index_value(*id);
+
+                    let unlock = (self.player_data.unlocks.mounts[index as usize] & value) == 0;
+
+                    if unlock {
```

Comment:
You can simplify this code due to the nature of bitwise XOR, which should always toggle the value right? (obv we still need the check for other code that depends on this condition like aether currents)